dot/state/storage.go

```diff
@@ -74,10 +74,18 @@ func NewStorageState(db chaindb.Database, blockState *BlockState, t *trie.Trie)
 	}, nil
 }
 
-func (s *StorageState) pruneStorage() { //nolint
-	// TODO: when a block is finalized, delete non-finalized tries from DB and mapping
-	// as well as all states before finalized block
-	// TODO: pruning options? eg archive, full, etc
+func (s *StorageState) pruneKey(hash common.Hash) {
```

Comment:
the hash that's passed into this function is the block hash, you need to find the `StateRoot` for that block, since `s.tries` stores tries by `StateRoot`

you can do this by getting the header:
```
	header, err := s.blockState.GetHeader(hash)
	if err != nil {
		return nil, err
	}
```

then using `s.tries[&header.StateRoot]`